R/create_landscape_covariates.R

```diff
@@ -0,0 +1,345 @@
+#'      be generates `c("mrvbf", "dah", "landform")`.
+#' @param sieve_size a `numeric`value to define the number of pixels to be grouped during the sieving process.
+#'      default is `10`.
+#' @param dah_threshold a `numeric`value to define threshold at which diurnal anisotrophic heating (dah) will
+#'      define catergories. Default value is `10`.
+#' @param saga_param named list. Parameters that are fed to SAGA MRVBF function. IF supplied it must contain
+#'      all parameters specified in the SAGA command.
+#'
+#' @return path to the output directory where files are written (invisibly).
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' #--- create all SAGA covariates ---#
+#' create_samplr_covariate(
+#'     dtm = dtm = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_rel,"25m","dem.tif"),
+#'     saga_path = find_saga_path()[3]
+#'     out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel)
+#'}
+create_landscape_covariates <- function(dtm = dtm,
+                                     saga_path = NULL,
+                                     out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel,
+                                     layers = c("mrvbf", "dah", "landform"),
+                                     sieve_size = 10,
+                                     dah_threshold = 0.2,
+                                     saga_param = list(
+                                       T_SLOPE = 64, TPCTL_V = 6, T_PCTL_R = 2,
+                                       P_SLOPE = 4.0, P_PCTL = 3.0, UPDATE = 1,
+                                       CLASSIFY = 1, MAX_RES = 100
+                                     )) {
+  # testing lines
+  # dtm = fs::path(PEMprepr::read_fid()$dir_1020_covariates$path_rel,"25m","dem.tif")
+  # saga_path = "C:/Programs/saga-9.2.0_x64/saga-9.2.0_x64/saga_cmd.exe"
+  # out_dir = PEMprepr::read_fid()$dir_1020_covariates$path_rel
+  #
+  # create_samplr_covariates(dtm, saga_path)
+  # layers = c("mrvbf", "dah", "landform")
+  # sieve_size = 10
+  # dah_threshold = 0.2
+  # saga_param = list(T_SLOPE = 64, TPCTL_V = 6, T_PCTL_R = 2,
+  #                  P_SLOPE = 4.0, P_PCTL = 3.0, UPDATE = 1,
+  #                  CLASSIFY = 1, MAX_RES = 100)
+  # end testing lines
+
+
+
+  #--- dtm ---#
+
+  if (inherits(dtm, c("character"))) {
+    if (!file.exists(fs::path(dtm))) {
+      cli::cli_abort("{.var dtm} must point to an existing dtm file")
+    } else {
+      dtm <- terra::rast(dtm)
+    }
+  }
+
+  #--- get resolution of dtm ---#
+  rn <- terra::res(dtm)[1]
+
+  #--- SAGA ---#
+
+  if (!is.null(saga_path)) {
+    check_saga(saga_path)
+    cli::cli_alert_success(
+      "Your SAGA connection has been succesfuly set up. Using the {.var {saga_path}}"
+    )
+  } else {
+    check_saga()
+    cli::cli_abort("{.var saga_path} must point to the saga_path location on your computer.
+                   Please check your program files or equivalent to locate the saga_path.exe file")
+  }
```

Comment:
I think this can be greatly condensed, since this logic is already in `check_saga()`

```suggestion
check_saga(saga_path)
```

Reply:
Ditto for the same code in `create_covariates()`